### PR TITLE
project-todos Caroline Byström

### PR DESCRIPTION
**Pull request by Frivridna**
Link to my deployed project:
https://your-todo-app.netlify.app/

Things I'd like to have clarified or explained in more detail:
How to work with React datepicker.

### CODE REVIEW

File: code/src/components/AddToDoForm.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+  const [newTodo, setNewTodo] = useState('')
+  const [category, setCategory] = useState('')
+
+  const onCategoryChange = (e) => {
+    setCategory(e.target.value)
+  }
+
+  const onFormSubmit = (event) => {
+		event.preventDefault()
```
Comment (Sofarz) [inline code/src/components/AddToDoForm.js:26]: Check indentation here :)

File: code/src/components/AddToDoForm.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+          id,
+          description: newTodo,
+          category,
+          isComplete: false,
+        })
+      )
+      setNewTodo('')
+      setCategory('')
+    }
+	}
```
Comment (Sofarz) [inline code/src/components/AddToDoForm.js:40]: Nicely done! 💯

File: code/src/components/AddToDoForm.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  border-radius: 50%;
+  margin-bottom: 40px;
+  background-color: #216dff;
+  color: #d5c2df;
+  width: 120px;
+  height: 120px;
+  transition: transform .5s;
+  text-transform: uppercase;
+  &:hover {
+    transform: scale(1.1);
```
Comment (Sofarz) [inline code/src/components/AddToDoForm.js:114]: Way to go on using styled components with hover effects and animations!

File: code/src/components/Counter.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  const onCheckAll = () => {
+    dispatch(todos.actions.markAllTodos())
+  }
+  
+  const completedTodos = allToDos.filter((todo) => todo.isComplete)
+
+  return (
+    <Container>
+      <CompletedTodos>
+        {completedTodos.length} / {allToDos.length}
```
Comment (Sofarz) [inline code/src/components/Counter.js:21]: Not sure what is considered best practice, but you could add the .length in const allTodos and const completedTodos instead to keep the code inside the return cleaner.

File: code/src/components/ZeroTodos.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+`;
+
+const DoubleTick = styled.img`
+  width: 40px;
+  height: 40px;
+  position: absolute;
+  top: -40px;
+  transform-origin:110% center; 
+  animation:rotate 4s linear;
+  @keyframes rotate { 100% { transform:rotate(-360deg); } }
```
Comment (Sofarz) [inline code/src/components/ZeroTodos.js:36]: 🥇 💯

**Review by Sofarz — APPROVED** [review]
Great job on this weeks project! Really nice work with all the extra features with the categories, dates and animations 👍  

Regarding the styling - maybe you could make sure that the categories are not showing when they are empty so that it doesn't take any unnecessary space on the users phone for example? Just a thought :)

File: code/src/components/AddToDoForm.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+          isComplete: false,
+        })
+      )
+      setNewTodo('')
+      setCategory('')
+    }
+	}
+
+  return (
+    <>
```
Comment (KarolCislowski) [inline code/src/components/AddToDoForm.js:43]: you have a single form here so you shouldn't use <></> here 🤓

File: code/src/components/TodoHome.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+            <DeleteOutlinedIcon 
+              onClick={() => dispatch(todos.actions.removeToDo(todo.id))} />
+          </CheckboxContainer>
+        ))}
+      </TodoList>
+    </Container>
+  )
+}
+
+export default TodoHome
```
Comment (KarolCislowski) [inline code/src/components/TodoHome.js:38]: all your sections have the same logic. You should create 1 template where you would pass the category name as a prop and reuse it.

File: code/src/components/AddToDoForm.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+        </UserInputTextarea>
+          <AddButton
+            type='submit'>
+              add todo
+            <img src="/assets/plus.svg" alt="plus sign on button"/>
+          </AddButton>
+          <Label htmlFor="category">
+            Choose category
+          </Label>
+          <CustomSelect
```
Comment (KarolCislowski) [inline code/src/components/AddToDoForm.js:62]: I think the category selection should be before the add button. 🤓

**Review by KarolCislowski — APPROVED** [review]
Nice app but remember to create reusable components if there is a lot of repeated code.

**Pull request closed without merging**